src/atomate2/vasp/flows/eos/base.py

```diff
@@ -0,0 +1,207 @@
+    number_of_frames : int
+        Number of strain calculations to do for EOS fit, default = 6
+    postprocessor : .job
+        optional postprocessing step
+    """
+
+    name: str = "EOS Maker"
+    relax_maker: Maker = field(default_factory=EosDoubleRelaxMaker)
+    deformation_maker: Maker = field(
+        default_factory=lambda: DeformationMaker(
+            copy_vasp_kwargs={"additional_vasp_files": ("WAVECAR",)}
+        )
+    )
+    static_maker: Maker | None = None
+    linear_strain: tuple[float, float] = (-0.05, 0.05)
+    number_of_frames: int = 6
+    postprocessor: Job = postprocess_EOS
+
+    def make(self, structure: Structure, prev_dir: str | Path | None = None):
+        """
+        Run an EOS VASP job.
+
+        Parameters
+        ----------
+        structure : Structure
+            A pymatgen structure object.
+        prev_dir : str or Path or None
+            A previous VASP calculation directory to copy output files from.
+        """
+        jobs: dict[str, list[Job]] = {"relax": [], "static": [], "postprocess": []}
+
+        relax_flow = self.relax_maker.make(structure=structure, prev_dir=prev_dir)
+        relax_flow.name = "EOS equilibrium relaxation"
+        equil_props = {
+            "relax": {
+                "E0": relax_flow.output.calcs_reversed[0].output.energy,
+                "V0": relax_flow.output.calcs_reversed[0].output.structure.volume,
+            }
+        }
+
+        if self.static_maker:
+            equil_static = self.static_maker.make(
+                structure=relax_flow.output.structure,
+                prev_dir=relax_flow.output.dir_name,
+            )
+            equil_static.name = "EOS equilibrium static"
+            equil_props["static"] = {
+                "E0": equil_static.output.calcs_reversed[0].output.energy,
+                "V0": equil_static.output.calcs_reversed[0].output.structure.volume,
+            }
+
+        strain_l, strain_delta = np.linspace(
+            self.linear_strain[0],
+            self.linear_strain[1],
+            self.number_of_frames,
+            retstep=True,
+        )
+        # Cell without applied strain already included from relax/equilibrium
+        # steps. Perturb this point (or these points) if included
+        zero_strain_mask = np.abs(strain_l) < 1.0e-15
+        if np.any(zero_strain_mask):
+            nzs = len(strain_l[zero_strain_mask])
+            shift = strain_delta / (nzs + 1.0) * np.linspace(-1.0, 1.0, nzs)
+            strain_l[np.abs(strain_l) < 1.0e-15] += shift
+
+        insert_equil_index = np.searchsorted(strain_l, 0, side="left")
+        deformation_l = [(np.identity(3) * (1 + eps)).tolist() for eps in strain_l]
+
+        for ideformation, deformation in enumerate(deformation_l):
+            # Doubly ensure that relaxations are done at fixed volume --> ISIF = 2
+            deform_relax_maker = update_user_incar_settings(
+                flow=self.deformation_maker, incar_updates={"ISIF": 2}
```

Comment:
It should be fine to just set this on the deformation maker input set. If users really want to override it then so be it.

This will also help keep the workflow generalisable.

Reply:
I would also like to add that ISIF=4 is a common setting for equations of state as well!